src/dateparser/dateparser.js

```diff
@@ -82,6 +108,7 @@ angular.module('ui.bootstrap.dateparser', [])
     }
 
     format = $locale.DATETIME_FORMATS[format] || format;
+    format = format.replace(SPECIAL_CHARACTERS_REGEXP, '\\$&');
```

Comment:
As per discussion between @chrisirhc and I on Slack, this turns out to be the best spot for modifying the format given how the iteration in `formatCodeToRegex` is implemented.